src/tools/crossgen2/Common/TypeSystem/Ecma/EcmaSignatureParser.cs

```diff
@@ -158,47 +175,60 @@ public bool IsFieldSignature
 
         public MethodSignature ParseMethodSignature()
```

Comment:
ParseMethodSignature is reentrant for creation of FunctionPointerType parameters, the current try/finally based approach loses modifier data.

Reply:
ALSO, when constructing FunctionPointerType instances, the custom modifier data of the method signature must not contribute to the MethodSignature contained within the FunctionPointerType. (The current PR allows that to happen.)

Reply:
The reentrancy issue is fixed, the signature in the type is still contaminated, need a fix.

Reply:
This is fixed as well